### DISCUSSION TITLE
Additional packages per-instance

### DISCUSSION THREAD

**Discussion by YuriyShestakov** [jlentink/aem#43] [Ideas]
Hi Jason! 

AEM CLI and AEM Commerce.

For the AEM commerce framework integration, we need to install CIF packages ("far" files) the way described [here](https://experienceleague.adobe.com/docs/experience-manager-cloud-service/content/content-and-commerce/storefront/developing/develop.html?lang=en#local-setup) (point 4) for our local AEM Cloud SDK instance. We don't want to add such packages to our pom.xml as they work automatically in non-local AEMaaCS environments.

It is not possible to do it with the "additional packages" functionality, as the feature is not separated for author & publisher. 
If I put those files manually (or by changing `additiontalPackages` + restarting each instance), the addon is getting removed after the next restart (when the `additionalPackages` array is empty) so everything stops working.

As a workaround, I tried to the security restrictions for the "install" folder to prevent file deletions in this folder (using the `chmod` command), but this is not working as aem-cli stops due to an error if it couldn't remove files from the directory. So I was forced to write my own script for starting aem which is copying those files to the install folder after `aem start`.

It would be good if we have these additional features in the AEM CLI:
* Specifying additional packages by relative path (we can add necessary files to git to the parent project folder or to the `instances` folder, as the automatic downloading from software distribution is a bit tricky).
* The ability to use the `additionalPackages` feature per-instance for any of `[[instance]]` sections in aem.toml
* Another one, we can instead somehow leverage the [`static/repository/install` folder functionality](https://helpx.adobe.com/experience-manager/kb/HowToInstallPackagesUsingRepositoryInstall.html) within aem-cli, but I am not sure how it can be done properly to make using it comfortably. Thinking now about something like having a `./instances/local-author/crx-quickstart/static/repository/install` folder in the source code, but the AEM CLI likely won't unpack to it if such a folder (local-author) already exists. Maybe this can also be updated in the cli code?

What do you think?

**Comment by jlentink** [comment]
Looks doable. Let's work together to build it in!
